cmd/web-handlers.go

```diff
@@ -667,9 +667,13 @@ next:
 	for _, objectName := range args.Objects {
 		// If not a directory, remove the object.
 		if !HasSuffix(objectName, SlashSeparator) && objectName != "" {
-			// Check for permissions only in the case of
-			// non-anonymous login. For anonymous login, policy has already
-			// been checked.
+			// Get the retention/lock info to check if it is enabled or not
+			info, ok := globalBucketObjectLockConfig.Get(args.BucketName)
+			bucketObjectLockEnabled := false
+			if ok && info.Mode != "" {
```

Comment:
I think you should check if the bucket was created with lock enabled, not if there is a default lock set on the bucket. need to check that.

Reply:
ok

Reply:
@kannappanr,
When we talked, my point was that the return value "ok" from `globalBucketObjectLockConfig` exactly doing what you were asking and that depending on solely on "ok", we'd loose to detect the case where there is a lock set but no locking configuration has yet been enabled. So, that was the logic behind using `ok && Mode != ""`
However, then you explained the Locking and Retention feature and how it works, what is the expected behavior, etc. 
Please correct me if I am wrong; but it looks like the overall logic in `RemoveObject function is somewhat quite not right. That is, we are talking about a change larger in scope than what issue #9410 complains about.
So, more clarification would definitely help if we would like to fix the overall logic instead of trying to address the specific delete problem for anonymous users explained in #9410.

Reply:
@kannappanr, 
I tested the fix after your input on the expected behavior and found out that for the delete function to be allowed to run for users who has a policy with `BypassGovernanceRetentionAction`, the HTTP header is supposed to come populated with `"X-Amz-Bypass-Governance-Retention"`. This needs to happen on the front end side, when http request is created, right?
You also said that we have a plan to get rid of the `bypass retention`.
Does it mean I need to ignore it and go ahead with the delete action after checking user's policy?

Reply:
@kannappanr,
Never mind. I figured it out.

Reply:
Done.